print-bgp.c

```diff
@@ -2506,6 +2522,60 @@ bgp_attr_print(netdissect_options *ndo,
             len -= 12;
         }
         break;
+    case BGPTYPE_BGPSEC_PATH:
+    {
+        uint16_t sblen, splen, siglen;
+        u_int idx;
+
+        splen = GET_BE_U_2(tptr);
+        idx = 0;
+
+        if (splen < 8) {
+            ND_PRINT(" [secure path too short]");
+            tptr += splen;
+            break;
+        }
+
+        ND_PRINT("\n\t    Secure Path Length: %u", splen);
+
+        idx += 2;
+        splen -= 2;
+        /* parse secure path segments */
+        while (splen > 0) {
```

Comment:
Insecure loop:
If Secure Path Length == 9 (invalid file), at the end of the loop `splen` will have the following values:
```
1
65531
65525
65519
...
```
Thus the `(splen > 0)` condition is not sufficient.
Before the loop is needed:
```
        if (splen == 0 || splen % 6) {
            ND_PRINT(" [invalid total segments len %u]", splen);
            break;
        }
```
and the condition of the loop should be `(splen != 0)` to make
clearer what we want to test.

Reply:
Thanks, good catch! I will make changes accordingly.

Reply:
Sadly, due to variable length of signature segments, the same cannot be applied there. But I added a check for minimum length.